ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/DbDispatcher.scala

```diff
@@ -110,15 +112,18 @@ object DbDispatcher {
       metrics: Metrics,
   )(implicit loggingContext: LoggingContext): ResourceOwner[DbDispatcher with ReportsHealth] =
     for {
-      hikariDataSource <- HikariDataSourceOwner(
-        dataSource,
-        serverRole,
-        connectionPoolSize,
-        connectionPoolSize,
-        connectionTimeout,
-        Some(metrics.registry),
-      )
-      connectionProvider <- DataSourceConnectionProvider.owner(hikariDataSource)
+      wrappedDataSource <-
+        if (dataSource.isInstanceOf[MDataSource]) ResourceOwner.successful(dataSource)
```

Comment:
hack to make it happen, landing it would require some refactoring here to allow polymorphism to trigger wrapping with HikariCP